### PR TITLE
[X86] Add USER_MSR instructions.

### PR DESCRIPTION
**Pull request by FreddyLeaf**
For more details about this instruction, please refer to the latest ISE document: https://www.intel.com/content/www/us/en/develop/download/intel-architecture-instruction-set-extensions-programming-reference.html

### CODE REVIEW

File: clang/lib/Headers/x86gprintrin.h [hunk 1]
```diff
@@ -20,6 +20,11 @@
 #include <uintrintrin.h>
 #endif
 
+#if !(defined(_MSC_VER) || defined(__SCE__)) || __has_feature(modules) ||      \
+    defined(__UINTR__)
```
Comment (phoebewang) [inline clang/lib/Headers/x86gprintrin.h:24]: __USERMSR__

File: llvm/CMakeLists.txt [hunk 2]
```diff
@@ -922,8 +922,6 @@ endif()
 
 include(HandleLLVMOptions)
 
-######
-
```
Comment (phoebewang) [inline llvm/CMakeLists.txt:926]: Unrelated change

File: llvm/include/llvm/TargetParser/X86TargetParser.def [hunk 3]
```diff
@@ -241,6 +241,7 @@ X86_FEATURE       (SM3,             "sm3")
 X86_FEATURE       (SM4,             "sm4")
 X86_FEATURE       (AVXVNNIINT16,    "avxvnniint16")
 X86_FEATURE       (EVEX512,         "evex512")
+X86_FEATURE       (USERMSR,            "usermsr")
```
Comment (phoebewang) [inline llvm/include/llvm/TargetParser/X86TargetParser.def:244]: Make it aligned above

File: llvm/test/MC/Disassembler/X86/usermsr-64.txt [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+# INTEL: urdmsr r9, 123
+0xc4,0xc7,0x7b,0xf8,0xc1,0x7b,0x00,0x00,0x00
+
+# ATT:   urdmsr %r9, %r9
+# INTEL: urdmsr r9, r9
+0xf2,0x45,0x0f,0x38,0xf8,0xc9
+
+# ATT:   urdmsr %r9, %r9
+# INTEL: urdmsr r9, r9
+0xf2,0x4d,0x0f,0x38,0xf8,0xc9
```
Comment (phoebewang) [inline llvm/test/MC/Disassembler/X86/usermsr-64.txt:14]: Why different encodings decode to the same instruction? Should avoid to use the same register?

File: llvm/lib/Target/X86/X86.td [hunk 5]
```diff
@@ -325,6 +325,8 @@ def FeatureTSXLDTRK : SubtargetFeature<"tsxldtrk", "HasTSXLDTRK", "true",
                                        "Support TSXLDTRK instructions">;
 def FeatureUINTR : SubtargetFeature<"uintr", "HasUINTR", "true",
                                     "Has UINTR Instructions">;
+def FeatureUSERMSR : SubtargetFeature<"usermsr", "HasUSERMSR", "true",
+                                    "Support USERMSR instructions">;
```
Comment (phoebewang) [inline llvm/lib/Target/X86/X86.td:329]: indent

File: llvm/test/MC/Disassembler/X86/usermsr-64.txt (hunk 4 above)
Comment (FreddyLeaf) [reply llvm/test/MC/Disassembler/X86/usermsr-64.txt:14]: All legacy Instructions with R64 supported could be judged as "software WIG". So this is a WIG instruction, too. Same as below:
```
$ echo "0xf3,0x41,0x0f,0xc7,0xf0" | llvm-mc --disassemble
        .text
        senduipi        %r8
$ echo "0xf3,0x49,0x0f,0xc7,0xf0" | llvm-mc --disassemble
        .text
        senduipi        %r8
```
Will add a comment.

File: llvm/lib/Target/X86/X86.td (hunk 5 above)
Comment (FreddyLeaf) [reply llvm/lib/Target/X86/X86.td]: 3e32e495d8fa56ad3770769a5d4559ea0a41af96

File: llvm/test/MC/Disassembler/X86/usermsr-64.txt (hunk 4 above)
Comment (FreddyLeaf) [reply llvm/test/MC/Disassembler/X86/usermsr-64.txt:15]: 3e32e495d8fa56ad3770769a5d4559ea0a41af96

File: llvm/include/llvm/TargetParser/X86TargetParser.def (hunk 3 above)
Comment (FreddyLeaf) [reply llvm/include/llvm/TargetParser/X86TargetParser.def]: 3e32e495d8fa56ad3770769a5d4559ea0a41af96

File: llvm/CMakeLists.txt (hunk 2 above)
Comment (FreddyLeaf) [reply llvm/CMakeLists.txt]: 3e32e495d8fa56ad3770769a5d4559ea0a41af96

File: clang/lib/Headers/x86gprintrin.h (hunk 1 above)
Comment (FreddyLeaf) [reply clang/lib/Headers/x86gprintrin.h]: 3e32e495d8fa56ad3770769a5d4559ea0a41af96

**Review by KanRobert — APPROVED** [review]
LGTM

**Review by phoebewang — APPROVED** [review]
